2_adventures/001/lenn4rd/03_raindrops_exercise/raindrops.rb

```diff
@@ -0,0 +1,25 @@
+class Fixnum
+  def prime_factor_of?(number)
+    number % self == 0
+  end
+end
+
+class Raindrops
+  PRIME_FACTORS = [3, 5, 7]
+
+  SOUNDS_FOR_FACTORS = {
+    '3' => 'Pling',
+    '5' => 'Plang',
+    '7' => 'Plong'
+  }
+  
+  def self.convert(number)
+    PRIME_FACTORS.each_with_object(sound = '') do |factor|
+      sound << SOUNDS_FOR_FACTORS[factor.to_s] if factor.prime_factor_of?(number)
+    end
+    
+    return number.to_s if sound.empty?
+    
+    sound
```

Comment:
You're right, it takes some time to get used to those logic statements. After I read your first comment, I used brackets to find the order in which Ruby evaluates those statements. It makes sense now that they're read left-to-right but I still think even for an experienced developer they'd take a few seconds to understand. I second your objection to the colon and question mark for the ternary operator. Actually I think I never used it _without_ one of those `question?` methods.

I'll gladly take some time to review your contribution. I generally try to pay someone back equally in the open source world. ;)